src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -1503,4 +1547,18 @@ void report(Event evt) {
       reporter.handle(evt);
     }
   }
+
+  /**
+   * A simple value class combining a hash of the tool inputs (and their digests) as well as a set
+   * of the relative paths of all tool inputs.
+   */
+  private static final class ToolSignature {
```

Comment:
Hi! I've been experimenting with remote persistent workers in a setting where the remote execution backend has zero knowledge of the tools. I ran into the issue of the backend not knowing how to actually run the tool. To overcome this, I added the actual command (with `--persistent_worker`) called by Bazel to `ToolSignature`. Implementation here: https://github.com/wiwa/bazel/commit/07d40cef1019dc510aacf49de22ae39831bd647e#diff-5556e28152f91eac5403663774979225b86e64133da164366b56f91d1daa641cR1548 

I then pass it to the backend via another entry in `platformAdditionalProperties` called `persistentWorkerCommand`: https://github.com/wiwa/bazel/commit/07d40cef1019dc510aacf49de22ae39831bd647e#diff-5556e28152f91eac5403663774979225b86e64133da164366b56f91d1daa641cR487

Do you think this PR could also support this functionality? Or would it have to be a different change?

Reply:
I propose that we not risk delaying this change by complicating it further. Let's file additional issues or pull requests for enhancements.

Reply:
Yep, I agree. Was not sure if it would particularly delay it, but the answer seems like "yes it would".

Reply:
@wiwa That sounds dangerous. If your backend doesn't know if it's running workers or not, it could end up with a lot of sometimes large processes hanging around.

Reply:
It's not that the backend doesn't know there are workers. It's that it doesn't need to know how to start specific worker processes. The backend is agnostic to the action which spins up a worker. I don't believe that is possible without addtional modification beyond the change in this PR. This is because Bazel persistent workers have more info to spawn the process than what is provided by a `persistentWorkerKey`.

Reply:
Example: a `Scalac` action calling `scalac.sh <various startup flags> --persistent_worker` to initialize the worker process before passing the "actual" compile args `<compilation flags> -cp <classpath> <files to compile>`. The remote execution backend would need to know the inputs/hash of `scalac.sh` and also what those startup flags are.